internal/resource/config/authenticationpolicies/policy/authentication_policies_policy_resource.go

```diff
@@ -0,0 +1,252 @@
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/authenticationpolicytreenode"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/id"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/resourcelink"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &authenticationPoliciesPolicyResource{}
+	_ resource.ResourceWithConfigure   = &authenticationPoliciesPolicyResource{}
+	_ resource.ResourceWithImportState = &authenticationPoliciesPolicyResource{}
+)
+
+// AuthenticationPoliciesPolicyResource is a helper function to simplify the provider implementation.
+func AuthenticationPoliciesPolicyResource() resource.Resource {
+	return &authenticationPoliciesPolicyResource{}
+}
+
+// authenticationPoliciesPolicyResource is the resource implementation.
+type authenticationPoliciesPolicyResource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type authenticationPoliciesPolicyModel struct {
+	Id                              types.String `tfsdk:"id"`
+	PolicyId                        types.String `tfsdk:"policy_id"`
+	Name                            types.String `tfsdk:"name"`
+	Description                     types.String `tfsdk:"description"`
+	AuthenticationApiApplicationRef types.Object `tfsdk:"authentication_api_application_ref"`
+	Enabled                         types.Bool   `tfsdk:"enabled"`
+	RootNode                        types.Object `tfsdk:"root_node"`
+	HandleFailuresLocally           types.Bool   `tfsdk:"handle_failures_locally"`
+}
+
+func (r *authenticationPoliciesPolicyResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	schema := schema.Schema{
+		Description: "Manages an Authentication Policy",
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Optional:    true,
+				Description: "The authentication policy name. Name is unique.",
+			},
+			"description": schema.StringAttribute{
+				Optional:    true,
+				Description: "A description for the authentication policy.",
+			},
+			"authentication_api_application_ref": resourcelink.CompleteSingleNestedAttribute(true, false, false,
+				"Authentication API Application Id to be used in this policy branch. If the value is not specified, no Authentication API Application will be used."),
+			"enabled": schema.BoolAttribute{
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(true),
+				Description: "Whether or not this authentication policy tree is enabled. Default is true.",
+			},
+			"root_node": authenticationpolicytreenode.ToSchema(),
```

Comment:
I think the description may differ for this node vs the root node of a fragment, so we may need a parameter for that.